src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -408,6 +408,20 @@ public int runBlockingSystemCallUntilResult(Node currentNode, BlockingAction<Int
         }, blockingNativeCallUnblockingAction.get());
     }
 
+    @TruffleBoundary
+    public int runBlockingNFISystemCallUntilResult(Node currentNode, BlockingAction<Integer> action) {
+        assert Errno.EINTR.defined();
+        int EINTR = Errno.EINTR.intValue();
+
+        return runUntilResult(currentNode, () -> {
+            int result = action.block();
+            if (result == -EINTR) {
```

Comment:
What is `-EINTR` ?

Reply:
A special value to indicate the call failed with errno set to EINTR.

Reply:
Okay, maybe this would be more clear as `EINTR_FAIL = -EINTR`

Reply:
The only caller is Ruby, and they need to agree on a convention. I'm not sure extracting this to a constant and duplicating the constant is better.

Reply:
Done, I called it `INTERRUPTED`.